### ISSUE TITLE
Issues with two TCK tests

### ISSUE THREAD

**Issue by johnament** [eclipse/microprofile-health#71]
I'm struggling with these two tests

```
[ERROR]   EnforceQualifierTest>Arquillian.run:164->testFailureResponsePayload:57 expected [204] but found [200]
[ERROR]   NoProcedureSuccessfulTest>Arquillian.run:164->testSuccessResponsePayload:55 expected [200] but found [204]
```

As best as I can tell, they both mean that there are no checks, but for different reasons.  I cannot tell why they expect two different response codes.  There's potential data leakage if we leave it like this.

I would recommend that both tests look for 204's.

**Comment by heiko-braun** [comment]
204 means no procedures found, in contrary to 200. The enforcer test should find a procedure, but the otherone not. If both fail, something must be wrong with your setup, rather than the test themselves

**Comment by johnament** [comment]
You're correct, this is the output from two different test runs.  Sorry if that wasn't clear.

Why do you believe the enforcer test should find a health check?  My understanding of `@Health` is that it causes the check to not be discovered.

**Comment by heiko-braun** [comment]
Quite the contrary: @Health should be used for discovery

**Comment by heiko-braun** [comment]
Take a look at the readme

**Comment by johnament** [comment]
Then why is `NoProcedureSuccessfulTest` expecting a 200 instead of a 204?

**Comment by heiko-braun** [comment]
ok, I see what's going on. The above description I supplied are not valid anymore. With #3 (See Scott changes), the the way to identify the case of "no procedures installed" is no longer `204`, but `200` with an empty body instead.

So if your impl still returns `204` for the `NoProcedureSuccessfulTest` then it probably needs to be updated to reflect the changes in #3. The other test `EnforceQualifierTest` should return `200` with empty body as well (no procedures should be found when the annotation is not present), but apparently this wasn't updated with #3. We need to fix that assertion.

**Comment by heiko-braun** [comment]
@johnament I've merged #72 to correct the assertion

**Comment by johnament** [comment]
Ok, that looks better.

#70 is my last issue as of now.
